chaincode/compute_plan.go

```diff
@@ -404,9 +408,88 @@ func UpdateComputePlanState(db *LedgerDB, ComputePlanID, tupleStatus, tupleKey s
 	if err != nil {
 		return err
 	}
-	if cp.UpdateStatus(tupleStatus) {
-		db.AddComputePlanEvent(ComputePlanID, cp.State.Status)
+	statusUpdated := cp.UpdateStatus(tupleStatus)
+	doneModels, err := cp.HandleIntermediaryModel(db)
+	if err != nil {
+		return err
+	}
+	if statusUpdated || len(doneModels) != 0 {
+		db.AddComputePlanEvent(ComputePlanID, cp.State.Status, doneModels)
 		return cp.SaveState(db)
 	}
 	return nil
 }
+
+// ListModelIfIntermediary will reference the hash model if the compute plan ID
+// is not empty and if it's an intermediary model meaning without any children
+func ListModelIfIntermediary(db *LedgerDB, ComputePlanID, tupleKey, modelHash string) error {
+	if ComputePlanID == "" {
+		return nil
+	}
+	cp, err := db.GetComputePlan(ComputePlanID)
+	if err != nil {
+		return err
+	}
+	if !cp.CleanModels {
+		return nil
+	}
+	allChildKeys, err := db.GetIndexKeys("tuple~inModel~key", []string{"tuple", tupleKey})
+	if err != nil {
+		return err
+	}
+	if len(allChildKeys) == 0 {
+		// If a tuple has no children it's concidered final and should not be
+		// listed in the index
+		return nil
+	}
+	cp.State.IntermediaryModel = append(cp.State.IntermediaryModel, modelHash)
+
+	return cp.SaveState(db)
+}
+
+// HandleIntermediaryModel is a function
```

Comment:
What would you think of the simpler

```suggestion
// HandleIntermediaryModel ...
```

Reply:
No that a "I will do it later" and, later on, an overview. Will be fixed.